R/navs-legacy.R

```diff
@@ -234,6 +256,25 @@ navbar_options <- function(
   )
 }
 
+navbar_options_apply_attribs <- function(navbar, navbar_options = NULL) {
+  if (is.null(navbar_options[["attribs"]])) {
+    return(navbar)
+  }
+
+  attribs <- navbar_options[["attribs"]]
+  navbar[[1]] <- rlang::exec(tagAppendAttributes, navbar[[1]], !!!attribs)
+
+  if ("data-bs-theme" %in% names(attribs)) {
+    # If you're setting this attribute directly, you know more about what you're
+    # doing than we do (we handle it for users via `type`). Also: the call to
+    # tagAppendAttributes ensures that `navbar[[1]]` is a tag object and has the
+    # attribs field.
+    navbar[[1]][["attribs"]][["data-bs-theme"]] <- attribs[["data-bs-theme"]]
```

Comment:
Now that I look into this, I'm not sure this is precisely what we want because of this behavior:

```r
> x <- list(a = 1, a = 2)
> x
$a
[1] 1

$a
[1] 2

> x[["a"]] <- 3
> x
$a
[1] 3

$a
[1] 2
```


Seems it would be better to remove the attribute before appending, like this:

```r
if ("data-bs-theme" %in% names(attribs)) {
  navbar[[1]][["attribs"]][["data-bs-theme"]] <- NULL
}

navbar[[1]] <- rlang::exec(tagAppendAttributes, navbar[[1]], !!!attribs)
```

Reply:
Great catch, also quite surprising...

``` r
x <- list(a = 1, a = 2)

x[["a"]] <- 3
x
#> $a
#> [1] 3
#> 
#> $a
#> [1] 2

x[["a"]] <- NULL
x
#> $a
#> [1] 2
```

Anyway, this will be entirely unnecessary if I just move some of the logic into `navs_bar_()`.